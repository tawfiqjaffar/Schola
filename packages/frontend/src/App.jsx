import React from 'react';
import { ThemeProvider } from '@material-ui/core';

import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware, compose } from 'redux';

import jwtDecode from 'jwt-decode';
import { setCurrentUser } from './actions/authActions';

import rootReducer from './rootReducer';
import Router from './components/common/Router';
import theme from './theme';

function App() {
  const store = createStore(
    rootReducer,
    compose(
      applyMiddleware(thunk),
      window.devToolsExtension ? window.devToolsExtension() : (f) => f
    )
  );

  if (localStorage.jwtToken) {
    store.dispatch(setCurrentUser(jwtDecode(localStorage.jwtToken)));
  }

  return (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router />
      </ThemeProvider>
    </Provider>
  );
}

export default App;
