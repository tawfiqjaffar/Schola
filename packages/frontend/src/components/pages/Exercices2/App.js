import React from 'react';

import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import Answer from './components/Answer/Answer';
import Result from './components/Result/Result';
import QuestionParagraph from './components/Question/Question.js';
import {withStyles} from '@material-ui/core/styles';
import styles from './app-style';
import questionsFactory from './models/factories/get-questions-factory';
import submitFactory from './models/factories/submit-questions-factory';
import axios from 'axios';

import CircularLoading from './components/CircularLoading';

import getQuestions from './models/services/questions/local/get-questions';
import {Link} from "react-router-dom";

class App extends React.Component {
    lastTimeButtonClicked = new Date().getTime();

    state = {
        questions: null,
        result: null,
        questionsAnswers: [],
        currentQuestionIndex: 0,
        step: 0,
        grades: [],
        score: 0,
    };

    componentDidUpdate(prevProps, prevState) {
        this.updateButtonsPos();

        setTimeout(() => {
            this.updateButtonsPos();
        }, 0);
    }

    componentDidMount() {
        window.onorientationchange = () => this.updateButtonsPos();
        window.onresize = () => this.updateButtonsPos();

        questionsFactory(false).then((value) => {
            this.setState({questions: value}, () => {
                this.updateButtonsPos();
            });
        });

        this.updateButtonsPos();

        setTimeout(() => {
            this.updateButtonsPos();
        }, 0);
        this.getAllGrades()
    }

    onNextClick = (e) => {
        const currentState = this.state;

        if (
            currentState.currentQuestionIndex === currentState.questions.length - 1 ||
            this.areButtonsAnimating()
        ) {
            return;
        }

        console.log(currentState.currentQuestionIndex);
        this.updatePage(++currentState.currentQuestionIndex);
    };

    onPrevClick = (e) => {
        const currentState = this.state;

        if (currentState.currentQuestionIndex === 0 || this.areButtonsAnimating()) {
            return;
        }

        this.updatePage(--currentState.currentQuestionIndex);
    };

    onSubmitClick = async () => {
        const currentState = this.state;

        this.setState({
            questions: null,
            questionsAnswers: [],
            currentQuestionIndex: 0,
        });

        let result = await submitFactory(
            null,
            currentState.questions,
            currentState.questionsAnswers
        );
        this.setState({
            result,
        });

        this.postGrade();
    };

    onTryAgainPressed = async () => {
        this.setState({
            questions: null,
            questionsAnswers: [],
            currentQuestionIndex: 0,
            result: null,
        });

        this.setState({
            questions: await getQuestions(),
        });
    };

    onAnswerSelected = (answerId) => {
        let clickedAnswerIndex = answerId;
        let currentState = this.state;
        const currentAnswers = this.state.questionsAnswers;
        currentAnswers[currentState.currentQuestionIndex] = clickedAnswerIndex;

        this.setState({
            userAnswerIndexes: currentAnswers,
        });
    };

    updateButtonsPos = () => {
        if (!this.questionsLoaded()) {
            return false;
        }

        if (this.state.step === 2) {
            let buttons = Array.from(
                document.getElementById('buttonsContainer').children
            );
            let mainContainer = document.getElementById('mainContainer');

            buttons.forEach((button) => {
                button.children[0].style.bottom = '0px';
            });

            const bottomPosition =
                mainContainer.clientHeight - mainContainer.scrollHeight + 20;

            buttons.forEach((button) => {
                button.children[0].style.bottom = bottomPosition + 'px';
            });
        }
    };

    updatePage = (questionIndex) => {
        this.lastTimeButtonClicked = new Date().getTime();
        document.getElementById('mainContainer').scrollTop = 0;
        this.setState({
            currentQuestionIndex: questionIndex,
        });
    };

    questionsLoaded = () => (this.state.questions !== null ? true : false);
    getCurrentQuestion = () =>
        this.state.questions[this.state.currentQuestionIndex].question;
    getCurrentAnswers = () =>
        this.state.questions[this.state.currentQuestionIndex].answers;
    isAnswerSelected = (answerIndex) =>
        this.state.questionsAnswers[this.state.currentQuestionIndex] ===
        answerIndex;
    shouldShowSubmit = () =>
        this.state.currentQuestionIndex === this.state.questions.length - 1 &&
        this.state.questionsAnswers[this.state.questions.length - 1] !== undefined;
    shouldShowNext = () =>
        this.state.currentQuestionIndex !== this.state.questions.length - 1 &&
        this.state.questionsAnswers[this.state.currentQuestionIndex] !== undefined;
    shouldShowPrev = () => false; // this.state.currentQuestionIndex !== 0;
    areButtonsAnimating = () => {
        const transitionTime = 600;
        const currentTime = new Date().getTime();

        return currentTime - transitionTime <= this.lastTimeButtonClicked;
    };

    getAllGrades = async (accessToken) => {
        let res;
        try {
            const response = await axios({
                method: 'get',
                url: 'http://localhost:8080/api/grade/subjectGrade/',
            });
            res = response.data;
        } catch (axiosErr) {
            res = axiosErr.response && axiosErr.response.data;
        }
        this.setState({grades: res.data});
    };

    postGrade = async () => {
        let res;
        try {
            const response = await axios({
                method: 'post',
                url: 'http://localhost:8080/api/grade/subjectGrade/',
                data: {grade: this.state.result.correctAnswers},
            });
            res = response.data;
        } catch (axiosErr) {
            res = axiosErr.response && axiosErr.response.data;
        }
        this.setState({grades: res.data});
    };

    render() {
        const {classes} = this.props;
        return (
            <Paper id="mainContainer" className={classes.paper} elevation={2}>
                {this.state.step === 0 ? (
                    <>
                        <div className="text-center">Bienvenue sur le quizz bg</div>
                        <div className="">
                            <Link to={'/create-quizz'}>
                                <div
                                    className="quizz1"
                                >
                                    Create Quizz
                                </div>
                            </Link>
                        </div>
                        <div className="row quizz-list">
                            <div
                                className="quizz1"
                                onClick={() => this.setState({step: 2})}
                            >
                                1 er quizz
                            </div>
                        </div>
                        <div className="history">
                            {this.state.grades.map((el) =>
                                <div>1 er quizz {el.grade}/10</div>
                            )}
                        </div>
                    </>
                ) : (
                    <>
                        <hr key={'horizontalLine'} width={'100%'}/>
                        {this.questionsLoaded() && this.state.result == null ? (
                            <div>
                                <div key={this.getCurrentQuestion()}>
                                    <QuestionParagraph
                                        question={this.getCurrentQuestion()}
                                        questionIndex={this.state.currentQuestionIndex + 1}
                                        questionsLength={this.state.questions.length}
                                    />

                                    <div className={classes.answerContainer}>
                                        {this.getCurrentAnswers().map((currentAnswer, index) => (
                                            <Answer
                                                answerIndex={index}
                                                key={this.getCurrentQuestion() + index}
                                                answer={currentAnswer}
                                                isSelected={this.isAnswerSelected(index)}
                                                onAnswerSelect={this.onAnswerSelected}
                                            />
                                        ))}
                                    </div>

                                    <div id="buttonsContainer">
                                        {this.shouldShowSubmit() ? (
                                            <Button
                                                variant="contained"
                                                className={classes.btnSubmit}
                                                onClick={this.onSubmitClick}
                                                color="primary"
                                            >
                                                Submit
                                            </Button>
                                        ) : null}

                                        {this.shouldShowNext() ? (
                                            <Button
                                                variant="contained"
                                                className={classes.btnNext}
                                                onClick={this.onNextClick}
                                                color="secondary"
                                            >
                                                Next
                                            </Button>
                                        ) : null}

                                        {this.shouldShowPrev() ? (
                                            <Button
                                                variant="contained"
                                                className={
                                                    this.shouldShowSubmit()
                                                        ? classes.btnPrevSubmit
                                                        : classes.btnPrev
                                                }
                                                onClick={this.onPrevClick}
                                                color="primary"
                                            >
                                                Prev
                                            </Button>
                                        ) : null}
                                    </div>
                                </div>
                            </div>
                        ) : this.state.result !== null ? (
                            <Result
                                result={this.state.result}
                                tryAgainPressed={this.onTryAgainPressed}
                            />
                        ) : (
                            <CircularLoading key={'loadingCircle'}/>
                        )}
                    </>
                )}
            </Paper>
        );
    }
}

export default withStyles(styles)(App);
