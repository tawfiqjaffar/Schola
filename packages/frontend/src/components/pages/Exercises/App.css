body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.quizz-container {
  margin-top: 50px;
  color: black;
  display: flex;
  justify-content: center;
}

.quizz-form {
  padding: 2cm;
  background-color: rgba(130, 197, 121, 0.61);
  color: black;
  border-radius: 16px;
  align-content: center;
  min-height: 30vh;
  line-height: 40px;
}

.quizz-submit {
  color: black;
  display: flex;
  justify-content: center;
}

.quizz-btn {
  margin-right: 8px !important;
  margin-left: 8px !important;
}

.MuiInputBase-input {
  color: white !important;
}

.popover-container {
  padding-right: 6px !important;
}
.msg-error {
  color: red;
  padding-right: 5000 !important;
  justify-content: center !important;
}

@import url(
  https://fonts.googleapis.com/css?family=Raleway:400,
  700,
  900,
  400italic,
  700italic,
  900italic
);

.error {
  text-align: center;
  justify-content: center !important;
  display: flex;
  flex-direction: column;
  margin-top: 100px;
  width: 100%;
}

.msg-error {
  animation: text-shadow 1.5s ease-in-out infinite;
  font-size: 5em;
  font-weight: 900;
  line-height: 1;
}

.msg-error:hover {
  animation-play-state: paused;
  justify-content: center !important;
}

@keyframes text-shadow {
  0% {
    transform: translateY(0);
    text-shadow: 0 0 0 #0c2ffb, 0 0 0 #2cfcfd, 0 0 0 #fb203b, 0 0 0 #fefc4b;
  }

  20% {
    transform: translateY(-1em);
    text-shadow: 0 0.125em 0 #0c2ffb, 0 0.25em 0 #2cfcfd, 0 -0.125em 0 #fb203b,
      0 -0.25em 0 #fefc4b;
  }

  40% {
    transform: translateY(0.5em);
    text-shadow: 0 -0.0625em 0 #0c2ffb, 0 -0.125em 0 #2cfcfd,
      0 0.0625em 0 #fb203b, 0 0.125em 0 #fefc4b;
  }

  60% {
    transform: translateY(-0.25em);
    text-shadow: 0 0.03125em 0 #0c2ffb, 0 0.0625em 0 #2cfcfd,
      0 -0.03125em 0 #fb203b, 0 -0.0625em 0 #fefc4b;
  }

  80% {
    transform: translateY(0);
    text-shadow: 0 0 0 #0c2ffb, 0 0 0 #2cfcfd, 0 0 0 #fb203b, 0 0 0 #fefc4b;
  }
}

@media (prefers-reduced-motion: reduce) {
  * {
    animation: none !important;
    transition: none !important;
  }
}