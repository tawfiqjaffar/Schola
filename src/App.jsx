import { ThemeProvider } from '@material-ui/core';
import React from 'react';
import NavigationBar from './components/common/NavigationBar';
import theme from './theme';

function App() {
  return (
    <>
      <ThemeProvider theme={theme}>
        <NavigationBar />
      </ThemeProvider>
    </>
  );
}

export default App;
