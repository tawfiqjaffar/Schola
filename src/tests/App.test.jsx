import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App component', () => {
  it('should render app and match snapshot', () => {
    const component = renderer.create(
      <App />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
