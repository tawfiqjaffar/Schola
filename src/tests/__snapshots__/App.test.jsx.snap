// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render app and match snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-title-1 MuiTypography-colorPrimary"
      href="/"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
        Schola
      </h4>
    </a>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-menuButton-2 makeStyles-buttonLogin-4 MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Log in
        </p>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-menuButton-2 makeStyles-buttonRegister-3 MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Register
        </p>
      </span>
    </button>
  </div>
</header>
`;
